homeassistant/components/lg_thinq/sensor.py

```diff
@@ -0,0 +1,590 @@
+    ThinQProperty.BATTERY_PERCENT: SensorEntityDescription(
+        key=ThinQProperty.BATTERY_PERCENT,
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key=ThinQProperty.BATTERY_PERCENT,
+    ),
+}
+DETERGENT_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.DETERGENT_SETTING: SensorEntityDescription(
+        key=ThinQProperty.DETERGENT_SETTING,
+        translation_key=ThinQProperty.DETERGENT_SETTING,
+    ),
+}
+DISH_WASHING_COURSE_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.CURRENT_DISH_WASHING_COURSE: SensorEntityDescription(
+        key=ThinQProperty.CURRENT_DISH_WASHING_COURSE,
+        translation_key=ThinQProperty.CURRENT_DISH_WASHING_COURSE,
+    )
+}
+DOOR_STATUS_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.DOOR_STATE: SensorEntityDescription(
+        key=ThinQProperty.DOOR_STATE,
+        translation_key=ThinQProperty.DOOR_STATE,
+    ),
+}
+FILTER_INFO_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.FILTER_LIFETIME: SensorEntityDescription(
+        key=ThinQProperty.FILTER_LIFETIME,
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        translation_key=ThinQProperty.FILTER_LIFETIME,
+    ),
+}
+HUMIDITY_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.CURRENT_HUMIDITY: SensorEntityDescription(
+        key=ThinQProperty.CURRENT_HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.CURRENT_HUMIDITY,
+    )
+}
+JOB_MODE_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.CURRENT_JOB_MODE: SensorEntityDescription(
+        key=ThinQProperty.CURRENT_JOB_MODE,
+        translation_key=ThinQProperty.CURRENT_JOB_MODE,
+    ),
+    ThinQPropertyEx.CURRENT_JOB_MODE_STICK_CLEANER: SensorEntityDescription(
+        key=ThinQProperty.CURRENT_JOB_MODE,
+        translation_key=ThinQPropertyEx.CURRENT_JOB_MODE_STICK_CLEANER,
+    ),
+    ThinQProperty.PERSONALIZATION_MODE: SensorEntityDescription(
+        key=ThinQProperty.PERSONALIZATION_MODE,
+        translation_key=ThinQProperty.PERSONALIZATION_MODE,
+    ),
+}
+LIGHT_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.BRIGHTNESS: SensorEntityDescription(
+        key=ThinQProperty.BRIGHTNESS,
+        translation_key=ThinQProperty.BRIGHTNESS,
+    ),
+    ThinQProperty.DURATION: SensorEntityDescription(
+        key=ThinQProperty.DURATION,
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        translation_key=ThinQProperty.DURATION,
+    ),
+}
+OPERATION_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.HOOD_OPERATION_MODE: SensorEntityDescription(
+        key=ThinQProperty.HOOD_OPERATION_MODE,
+        translation_key=ThinQProperty.OPERATION_MODE,
+    ),
```

Comment:
I know what you are thinking.
You might think that we'd better handle it as binary sensor.
But because the cloud sent it as string based on the profile spec as below.
- https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Hood/
We need to handle it as sensor.



R | POWER_ON | Hood On
-- | -- | --
R | POWER_OFF | Hood Off

Reply:
We will have more binary sensors right? How do they receive their state? I think we should explore the possibility to add them as binary sensor. I'm quite convinced that that is possible. So let's also split this one off. If we find out it's indeed going to be a big hassle, we can always decide to merge it in again

Reply:
HOOD_OPERATION_MODE has deleted. I'll prepare next binary_sensor PR. Thanks.